### ISSUE TITLE
Grid.sort() with column with classNameGenerator set causes javascript errors in grid connector

### ISSUE THREAD

**Issue by ggecy** [vaadin/vaadin-grid-flow#720]
When I use the `Grid.Column.setClassNameGenerator()` to set the class generator on column and then call the `Grid.sort()` method with that column in sort order I get the following JS errors when page renders in browser:
```
Uncaught TypeError: Cannot read property '_flowId' of undefined
    at HTMLElement.grid.cellClassNameGenerator (gridConnector.js?2614:940)
    at eval (vaadin-grid-styling-mixin.js?b20f:60)
    at Array.forEach (<anonymous>)
    at HTMLElement._generateCellClassNames (vaadin-grid-styling-mixin.js?b20f:55)
    at HTMLElement._updateItem (vaadin-grid.js?cd77:708)
    at HTMLElement.grid._updateItem (gridConnector.js?2614:437)
    at HTMLElement._getItem (vaadin-grid-data-provider-mixin.js?c382:182)
    at HTMLElement._updateScrollerItem (vaadin-grid.js?cd77:657)
    at HTMLElement.eval (vaadin-grid-scroller.js?2c8e:244)
    at HTMLElement._iterateItems (iron-list.js?1b01:607)
    at HTMLElement._assignModels (vaadin-grid-scroller.js?2c8e:241)
    at HTMLElement._update (iron-list.js?1b01:463)
    at HTMLElement._render (iron-list.js?1b01:550)
    at HTMLElement._accessIronListAPI (vaadin-grid-scroller.js?2c8e:307)
    at HTMLElement._render (vaadin-grid-scroller.js?2c8e:329)
    at HTMLElement._onAnimationEnd (vaadin-grid.js?cd77:757)
```

I defined 3 columns in java, but when I tried to debug the code the `<tr>` contains 4 `<td>` elements where the last one has the `_column` property undefined.

Using Google Chrome and Vaadin 14.0.0.rc6

**Comment by pekam** [comment]
Sounds like this might be already fixed by #717.
Please check if the issue reoccurs with 14.0.0.rc7 once it's released (should be tomorrow).

Still, it's weird that there's an extra `<tr>` element when using `sort()`. Unless you are using item details renderer as well?

**Comment by ggecy** [comment]
> Sounds like this might be already fixed by #717.
> Please check if the issue reoccurs with 14.0.0.rc7 once it's released (should be tomorrow).
> 
> Still, it's weird that there's an extra `<tr>` element when using `sort()`. Unless you are using item details renderer as well?

Hi, yes, there is item details renderer as well in this table. I have 3 columns defined, but there are 4 td elements inside table, see screenshots:

![extra_col1](https://user-images.githubusercontent.com/924873/61796177-2f2cd100-ae25-11e9-970e-6facc4a58fc0.png)

![extra_col2](https://user-images.githubusercontent.com/924873/61796186-3358ee80-ae25-11e9-8b51-066da63c40b7.png)

**Comment by pekam** [comment]
> It fails on the 4th cell, I guess this is the item details cell.

Yes, that's correct.

I'd guess that the issue occurs also without the `sort()` method and is caused only by the combination of cell class name generator and details renderer.

**Comment by ggecy** [comment]
> I'd guess that the issue occurs also without the `sort()` method and is caused only by the combination of cell class name generator and details renderer.

No, when I comment the initial grid.sort() call after I construct the grid, it works fine.

**Comment by pekam** [comment]
So it doesn't set any `cellClassNameGenerator` until sorting? In this case it still sounds like a combination of `cellClassNameGenerator` and `detailsRenderer`.

**Comment by ggecy** [comment]
The classNameGenerators are set on all columns before sorting, they just return nulls for unsorted columns and class name for sorted one.

**Comment by pekam** [comment]
Ah, ok. The issue indeed doesn't occur until the generator returns some values for a row with details cell. 

With this information I'm confident that we can close the issue as fixed by #717. Just bump your Vaadin version to rc7 as soon as it's out. :)
